os-audit/cos-auditd-logging.yaml

```diff
@@ -67,34 +51,38 @@ spec:
             memory: "10Mi"
             cpu: "10m"
       containers:
-      - name: fluentd-gcp-cos-auditd
+      - name: cos-auditd-fluent-bit
+        securityContext:
+          allowPrivilegeEscalation: false
+          capabilities:
+            drop:
+            - all
+            add:
+            - DAC_OVERRIDE
         env:
         - name: NODE_NAME
           valueFrom:
             fieldRef:
               apiVersion: v1
               fieldPath: spec.nodeName
-        image: gcr.io/stackdriver-agents/stackdriver-logging-agent:1.9.4
+        # Substitute these (manually or via envsubst). For example, run
+        # `CLUSTER_NAME=example-cluster CLUSTER_LOCATION=us-central1-a envsubst '$CLUSTER_NAME,$CLUSTER_LOCATION' < ${THIS_FILE:?} | kubectl apply -f -`
+        - name: CLUSTER_NAME
+          value: "$CLUSTER_NAME"
+        - name: CLUSTER_LOCATION
+          value: "$CLUSTER_LOCATION"
+        image: gke.gcr.io/fluent-bit@sha256:436f3b7a38522314dd3db22ae8187192d928763c29e94d04c0900b34f0ca0779 # v1.8.12-gke.16
```

Comment:
Note: I think I tried that once, but you could end up putting *any* tag in there, including one that doesn't exist. I suppose the tag is simply being ignored in favour of the hash (which makes sense). Therefore, including the tag in the path, rather than as a comment, doesn't actually ensure that the tag is correct. So it's effectively just a comment that doesn't look like a comment.

Reply:
That's good to know!